libfwupd/fwupd-enums.h

```diff
@@ -104,6 +104,7 @@ typedef enum {
  * @FWUPD_DEVICE_FLAG_MD_SET_VERFMT:		Set the device version format from the metadata if available
  * @FWUPD_DEVICE_FLAG_ADD_COUNTERPART_GUIDS:	Add counterpart GUIDs from an alternate mode like bootloader
  * @FWUPD_DEVICE_FLAG_NO_GUID_MATCHING:		Force an explicit ID match when adding devices to the device list
+ * @FWUPD_DEVICE_FLAG_PROXY_UPDATABLE:		Device is updatable but only matched when a proxy device is used
```

Comment:
I guess i'm not super sure about this.  Can't the plugin just only set the updateable flag when the `set_proxy` call is used?

Reply:
Maybe it's not a very good name; this is supposed to mean "_updatable enough so that it passes the FuInstallTask checks, but not updatable enough that it shows up as a first class update in fwupdmgr or gnome-software_". The previous name I had for this was `UPDATABLE_HIDDEN`.

Reply:
Two more ideas:
1.  Use `FWUPD_DEVICE_FLAG_IGNORE_VALIDATION` for this purpose.
2. `FWUPD_DEVICE_FLAG_IGNORE_CHECKS`

Reply:
I really want to describe the device, not describe what happens in the daemon. e.g. the device is updatable to the end user, but it doesn't need to manually update it as something else does. `UPDATABLE_HIDDEN` works in that idea, as does `UPDATABLE_PROXY` -- or `UPDATABLE_MASKED` or `UPDATABLE_PRIVATE` perhaps.

Reply:
I think of those you proposed my preference lands on hidden.  And specifically the comments around the flag should talk about how clients shouldn't do anything with it.

Another thought though - what about adding splitting to two classes of flags?  One being private device flags that clients shouldn't do anything with and one that they should?

The transition might be a bit ugly since clients can access some of the private ones, but we could make new functions like `fwupd_device_get_public_flags` and deprecate the `fwupd_device_get_flags`

Reply:
UPDATABLE_HIDDEN seems most obvious to me too. Changed.

> Another thought though - what about adding splitting to two classes of flags

Yes, I'm aware we only have 64 bits to use too. I'll have a think for a future PR. 1.5.x stuff tho.